polartools/diffraction.py

```diff
@@ -905,16 +941,26 @@ def plot_2d(
             detector=detector,
             monitor=monitor,
             normalize=normalize,
-            scale=scale,
             **kwargs,
         )
     fig = plt.figure()
     ax = fig.add_subplot(1, 1, 1)
     cmap = plt.get_cmap("rainbow")
+    if scale:
```

Comment:
You may want to consider using `np.nanpercentile` to setup the scale, it's very useful to get a useful scale if there's glitches in the data. Maybe something like:

```python
if scale == "auto":
    scale = (
        np.nanpercentile(1, dataz),
        np.nanpercentile(99, dataz)
    )
```